content/blog/marquee-1-0-0/index.Rmd

```diff
@@ -0,0 +1,217 @@
+
+## Use in ggplot2
+
+The number of people using this directly in grid is probably small. It is more likely that you access the functionality of marquee through higher level functions. Marquee provides two such functions aimed at making it easy to use marquee in ggplot2. The aim is to eventually move these into ggplot2 proper, but while we are in the initial phase of development they will stay in this package.
+
+### `geom_marquee()`
+
+The first function is (obviously) a geom. It is intended as a stand-in replacement for both `geom_text()` and `geom_label()`. As with `marquee_grob()` it works very unsurprisingly:
+
+```{r}
+library(ggplot2)
+# Add styling around the first word
+red_bold_names <- sub("(\\w+)", "{.red **\\1**}", rownames(mtcars))
+
+ggplot(mtcars) + 
+  geom_marquee(aes(x = mpg, y = disp, label = red_bold_names))
+```
+
+Apart from standard, but markdown-aware, `geom_text()` behaviour, the geom also gains a `width` aesthetic that allows you to turn on automatic soft wrapping of the text. In addition to this it gains a `style` aesthetic to finely control the style (more about styling below)
+
+### `element_marquee()`
+
+The second obvious use for marquee in ggplot2 is in formatting text elements. `element_marquee()` is a replacement for `element_text()` that does just that.
+
+```{r}
+ggplot(mtcars) + 
+  geom_point(aes(x = mpg, y = disp)) + 
+  ggtitle(md_text) + 
+  theme(plot.title = element_marquee(size = 8, width = unit(16, "cm")))
+```
+
+## Styling
+
+As alluded to above, marquee comes with a styling API that is reminiscent of CSS but completely its own. In some sense it takes the "simplicity over power" approach from markdown and applies it to styling.
+
+In marquee, each element type (e.g. a code block) has its own style. This style can be incomplete in which case it inherits the remaining specifications from the parent element in the document. As an example, the `em` element has the following default style `style(italic = TRUE)`, that is, take whatever style is currently in effect but set the text to italic. This is basically how CSS works as well, but CSS allows so much more which clutters both the API and implementation if all you ever want to do is format rich text.
+
+Apart from the direct inheritance of the marquee styling, it is also possible to use relative inheritance for numeric specifications (e.g. `lineheight = relative(2)` to double the current lineheight) or set sizes based on the current or root element font size (using `em()` and `rem()` respectively). Lastly, you can also mark a specification as "non-inheritable" using `skip_inherit()` in which case the inheritance moves up one level.
```

Comment:
I don't get what this means.

Reply:
if you set a value as `skip_inherit()` you say that any children should not inherit this value but instead the parent value. I'll try to come up with a better explanation